app.rb

```diff
@@ -18,28 +19,32 @@ def init
 def handle_event(event:, context:)
   init
 
-  record_results = []
-
+  records_to_process = []
+  # Parse records into array for parallel processing
   event["Records"]
     .select { |record| record["eventSource"] == "aws:kinesis" }
     .each do |record|
-      decoded_record = parse_record(record)
+      records_to_process << record
+    end
 
-      unless decoded_record && should_process?(decoded_record)
-        record_results << (decoded_record ? [decoded_record['id'], 'SKIPPING'] : [nil, 'ERROR'])
-        next
-      end
+  # Process records in parallel
+  record_results = Parallel.map(records_to_process, in_processes: 3) { |record| process_record(record) }
+end
 
-      record_results << process_record(decoded_record)
-    end
-  
-  return record_results
+def process_record record
+  decoded_record = parse_record(record)
+
+  unless decoded_record && should_process?(decoded_record)
+    return decoded_record ? [decoded_record['id'], 'SKIPPING'] : [nil, 'ERROR']
+  end
+
+  return store_record(decoded_record)
 end
 
 def parse_record record
   begin
     avro_data = record["kinesis"]["data"]
-  rescue KeyError => e
+  rescue *[KeyError, NoMethodError] => e
```

Comment:
Cool. Haven't used that syntax before. I like it

Reply:
haha yeah, any new syntax like that is from me thinking "I would do it this way in python, how can I make that work in ruby?". If you haven't before check it out, they call it "splat" in ruby (you can do ** to unpack hashes too), it's pretty useful when passing arguments